core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1011,6 +1014,30 @@ private[spark] object Utils extends Logging {
   }
 
   /**
+   * Convert a time parameter such as (50s, 100ms, or 250us) to microseconds for internal use. If
+   * no suffix is provided, the passed number is assumed to be in us.
+   */
+  def timeStringAsUs(str: String): Long = {
```

Comment:
Are the microseconds methods used anywhere? I'm not aware of anything that uses this resolution, but might have missed it. There aren't us methods in SparkConf.